datastructure/trie.py

```diff
@@ -0,0 +1,35 @@
+class Node:
+    def __init__(self, char):
+        self.char = char
+        self.children = {}
+        self.is_end = False
+
+
+class Trie:
+
+    def __init__(self):
+        self.root = Node('*')
+
+    def insert(self, word):
+        current = self.root
+
+        for char in word:
+            if char not in current.children:
+                current.children[char] = Node(char)
+            current = current.children[char]
+        current.is_end = True
+
+    def find(self, word):
```

Comment:
This is fine. another would be containsPrefix(prefix). Another to add would be allWords(prefix) that spits out all legit strings that start with prefix (i.e. if we search "ba" and we had words "baby", "bath", "barn" etc) it will return them all. 

This is the power of tries, it groups similar strings together through the nodes making its faster to find these types of things.